numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -1887,6 +1887,22 @@ gentype_reduce(PyObject *self, PyObject *NPY_UNUSED(args))
         mod = ((PyObjectScalarObject *)self)->obval;
         PyTuple_SET_ITEM(ret, 1, Py_BuildValue("NO", obj, mod));
     }
+    else if (obj && PyDataType_FLAGCHK((PyArray_Descr *)obj, NPY_LIST_PICKLE)) {
+        /* a structured dtype with an object in a field */
+        PyObject *arr = PyArray_FromScalar(self, NULL);
+        if (arr == NULL) {
+            return NULL;
+        }
+        PyArrayIterObject *it = (PyArrayIterObject *)PyArray_IterNew(arr);
+        if (it == NULL) {
+            return NULL;
+        }
+        mod = ((PyArray_Descr *)obj)->f->getitem(it->dataptr, arr);
```

Comment:
Rather than reusing the unrelated `mod` (module, line 1875), can we rename this to something else? `scalar`? `val`?

Reply:
A comment saying what python type this is would be handy too

Reply:
ok